src/parser/jobs/rdm/modules/ManaGauge.tsx

```diff
@@ -179,6 +158,34 @@ export class ManaGauge extends CoreGauge {
 		this.blackManaGauge.modify(amount.black)
 	}
 
+	private onApplyMagickedSwordPlay() {
+		if (this.activeGcdHook == null) {
+			// Buffs with stacks generate separate apply events for each stack with a single remove at the end.
+			// Make sure we only start hooking for actions effected by Magicked on the first apply event -- no duplicate hooks on the "reapply" events as the stacks go down
+			this.activeGcdHook = this.addEventHook(
+				filter<Event>()
+					.source(this.parser.actor.id)
+					.type('damage'),
+				this.onHitUnderMagickedSwordPlay
+			)
+		}
+	}
+
+	private onRemoveMagickedSwordPlay() {
+		if (this.activeGcdHook != null) {
+			this.removeEventHook(this.activeGcdHook)
+			this.activeGcdHook = undefined
+		}
+	}
+
+	//Honestly I don't really need to do anything here; but I needed something to hook for onApplyMagickedSwordPlay
+	//This code was pretty much taken as is from the BloodGauge handling for Blood Weapon
+	private onHitUnderMagickedSwordPlay(event: Events['damage']) {
+		if (event.cause.type === 'status') {
+			return
+		}
+	}
+
```

Comment:
Since we're not doing anything with the hook for attacks under Magicked Swordplay, we shouldn't include any of this.

However, we do need to do _something_ so that onGaugeSpender doesn't decrement gauge while Magicked Swordplay is in effect (easiest thing to do would be to check for hasStatus(MAGICKED_SWORDPLAY) at the top of the gauge spender method and bail early, would just need to make sure we don't have event ordering hijinx between when the action (cast) event happens and when the buff falls off on consuming the final stack.

Reply:
It appears as though the action happens before the status modification.  I've removed all of this, added a dependency on actors so I can check hasStatus and added a check to onGaugeSpender.  Below is an example of the ordering of events.

![image](https://github.com/user-attachments/assets/a85b682a-b517-42c4-8572-d9486bf7c121)